docs/knowledgebase/smart-contracts/index.md

```diff
@@ -1,31 +1,190 @@
+### Custom
+
+Substrate is not a platform where you are limited with what comes out of the box.
+
+We encourage further development of alternative smart contract platforms on top of the Substrate
+runtime. Use these pre-built modules to inform how you might design your own system or port over
+existing systems to work on a Substrate based chain.
+
+## Smart Contract Languages
+
+### ink!
+
+[ink!](ink-fundamentals) is a Rust-based eDSL for writing Wasm smart contracts specifically for the
+[Contracts module](#contracts-module). It is designed for correctness, conciseness and efficiency.
+
+## Smart Contracts vs Runtime Modules
+
+Substrate runtime modules and Substrate smart contracts are two different approaches to building
+"decentralized applications" using the Substrate framework.
+
+### Smart Contracts
+
+A traditional smart contract platform allows users to publish additional logic on top of some core
+blockchain logic. Since smart contract logic can be published by anyone, including malicious actors
+and inexperienced developers, there are a number of intentional safe guards built around these
+public smart contract platform.
+
+Some examples are:
+
+- **Fees**: Ensuring that contract developers are charged for the computation and storage they force
+  on the computers running their contract, and not allowed to abuse the block creators.
+- **Sandbox**: A contract is not able to modify core blockchain storage or the storage of other
+  contracts directly. It's power is limited to only modifying it's own state, and the ability to
+  make outside calls to other contracts or runtime functions.
+- **State Rent**: A contract takes up space on the blockchain, and thus should be charged for simply
+  existing. This ensures that people don't take advantage of "free, unlimited storage".
+- **Reversion**: A contract can be prone to have situations which lead to logical errors. The
+  expectations of a contract developer are low, so extra overhead is added to support reverting
+  transactions when they fail so no state is updated when things go wrong.
+
+These different overheads makes running contracts slower and more costly, but again, the "target
+audience" for contract development is different than runtime developers.
+
+Contracts can allow your community to extend and develop on top of your runtime logic without
+needing to go through all the craziness of proposals, runtime upgrades, etc... It may even be used
+as a testing grounds for future runtime changes, but done in a way that isolates your network from
+any of the growing pains or errors which may occur.
+
+**In summary**, Substrate Smart Contracts:
+
+- Are inherently safer to the network.
+- Have built in economic incentives against abuse.
+- Have computational overhead to support graceful failures in logic.
+- Have a lower bar to entry for development.
+- Enable fast pace community interaction through a playground to write new logic.
+
+### Runtime Modules
```

Comment:
should we call them pallets?

Reply:
Right. The whole article is reviewed to replace all module to pallet.